src/core/handle.ml

```diff
@@ -436,6 +436,7 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
     let (tm, ss) = time (handle_cmd ss) cmd in
     if Stdlib.(tm >= !too_long) then
       wrn cmd.pos "It took %.2f seconds to handle the command." tm;
+    begin match snd ss with None -> Meta.reset_key_counter () | _ -> () end;
```

Comment:
Why not doing this on line 436 in handle.ml or in line 62 in compile.ml (no need to test if there is a proof_data in both cases)?

Reply:
Ah so I think I do not understand your comment about `proof_data`, since there is no `proof_data` on line 436.

Reply:
Right. There is need for testing proof_date, contrary to what I said in my first comment. But after looking at the code in more details, I realized that there was no need to have and test a proof data. Does it make sense? Now I don't know where it is best to reset: in compile or in handle. What do you think?

Reply:
I think it's better to reset in handle, so that compile only wraps `handle_cmd` to take care of exceptions, and so we know clearly who does what. Does it make sense?

Reply:
Well reading again the code, not really since `handle` in compile already performs computation related to tactics.

Reply:
Maybe we should ask @ejgallego, since he may use `handle_cmd` to parse incrementally a file. For us, it doesn't matter much since we always compile files completeley.

Reply:
So what do you propose finally?

Reply:
We can place it in `compile`, to keep side effects closer to entry points (and avoid putting them deep into computations)